tornado-api/src/main/java/uk/ac/manchester/tornado/api/types/tensors/Shape.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2024, APT Group, Department of Computer Science,
+ * The University of Manchester.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.manchester.tornado.api.types.tensors;
+
+import java.util.Arrays;
+
+public record Shape(long... dimensions) {
+
+    public Shape(long... dimensions) {
+        this.dimensions = dimensions.clone();
+    }
+
+    public long[] dimensions() {
+        return dimensions.clone();
+    }
+
+    public int getRank() {
+        return dimensions.length;
+    }
+
+    public long[] getDimensions() {
+        return dimensions;
+    }
+
+    public int getSize() {
+        int size = 1;
+        for (long dim : dimensions) {
```

Comment:
You can use the Java 8 Stream API to perform this computation.

Reply:
fixed